src/core/main_task.c

```diff
@@ -63,6 +73,31 @@ void MainTask(void *pvParameters)
 {
     TickType_t xLastWakeTime = xTaskGetTickCount();
 
+    write_time = xTaskGetTickCount();
+
+    uint8_t eeprom_init_flag_addr = 0;
+    uint32_t operating_time_addr = 1;
+    uint32_t cycles_number_addr = 5;
+
+    // Check if EEPROM has been initialized
+    if (EEPROM_INIT_MAGIC_NUMBER == eeprom_read_byte(&eeprom_init_flag_addr)) {
+        // Not first boot, read EEPROM content
+        total_operating_time = eeprom_read_dword(&operating_time_addr);
+        cycles_number = eeprom_read_dword(&cycles_number_addr);
+#if DEBUG_MAIN
+        debug_print("[MAIN] EEPROM, tot: %u\r\n", total_operating_time);
+        debug_print("[MAIN] EEPROM, cn: %u\r\n", cycles_number);
+#endif
+    }
+    else {
+        //First boot ever
+        total_operating_time = 0;
+        cycles_number = 0;
+        taskENTER_CRITICAL();
+        eeprom_write_byte(&eeprom_init_flag_addr, EEPROM_INIT_MAGIC_NUMBER);
```

Comment:
Shouldn't we first initialize to zero the counters before writing the magic number & make it dword ?
```suggestion
eeprom_write_dword(operating_time_addr, 0);
eeprom_write_dword(cycles_number_addr, 0);
eeprom_write_dword(eeprom_init_flag_addr, EEPROM_INIT_MAGIC_NUMBER);
` ``